phoenicis-scripts/src/main/java/org/phoenicis/scripts/engine/implementation/TypedScriptEngine.java

```diff
@@ -0,0 +1,16 @@
+package org.phoenicis.scripts.engine.implementation;
+
+/**
+ * A typed script engine used by Phoenicis
+ *
+ * @param <T> The result type of the script
+ */
+public interface TypedScriptEngine<T> {
+    /**
+     * Evaluates the script with the given id and returns its result cast to the given type
+     *
+     * @param scriptId The id of the evaluated script
+     * @return The cast result of the evaluated script
+     */
+    T evaluate(String scriptId);
+}
```

Comment:
This class functions as an intermediate class for direct usage. It returns the result of an evaluated script (this time given by the script id) as an application type (e.g. `EngineSetting`, `Installer` etc.)